spyder/plugins/editor/widgets/tests/test_codeeditor.py

```diff
@@ -56,3 +57,56 @@ def test_editor_lower_to_upper(editorbot):
     widget.transform_to_uppercase()
     new_text = widget.get_text('sof', 'eof')
     assert text != new_text
+
+
+@pytest.mark.parametrize(
+        "input_text, expected_text, keys",
+             [
+              ('for i in range(2):\n    ',
+               'for i in range(2):\n\n    ',
+               [Qt.Key_Enter]),
+              ('myvar = 2 ',
+               'myvar = 2\n',
+               [Qt.Key_Enter]),
+              ('somecode = 1\nmyvar = 2 \nmyvar = 3',
+               'somecode = 1\nmyvar = 2 \nmyvar = 3',
+               [' ', Qt.Key_Up, Qt.Key_Up]),
+              ('somecode = 1\nmyvar = 2 ',
+               'somecode = 1\nmyvar = 2 ',
+               [Qt.Key_Left]),
+              ('"""This is a string with important spaces\n    ',
+               '"""This is a string with important spaces\n    \n',
+               [Qt.Key_Enter]),
+              ('somecode = 1\nmyvar = 2',
+               'somecode = 1\nmyvar = 2',
+               [' ', (Qt.LeftButton, 0)]),
+              ('somecode = 1\nmyvar = 2',
+               'somecode = 1\nmyvar = 2 ',
+               [' ', (Qt.LeftButton, 23)]),
+              ('a=1\na=2 \na=3',
+               'a=1\na=2 \na=3',
+               [(Qt.LeftButton, 6), Qt.Key_Up]),
+              ])
+def test_editor_rstrip_keypress(
+        editorbot, input_text, expected_text, keys):
+    """Test that whitespace is removed when leaving a line.
+    """
```

Comment:
Either fit this on one line, or have both ``"""`` on their own line, with the summary line in the middle.